### PR TITLE
Detailed rebranding Part 2

### PR DESCRIPTION
**Pull request by mbykhovtsev-ms**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id for toolchain build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=522913&view=results
- Pipeline build id for RPM build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=522980&view=results
- Pipeline build id for build image stage: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=523104&view=results

### CODE REVIEW

File: SPECS-EXTENDED/cassandra/Readme [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # apply pre-downloaded cassandra-build-cache which is
 # required for cassandra to build based on ant build.xml
 # Generate cassandra-build-cache when ever update to latest version.
-1. Boot up Mariner Core 2.0
+1. Boot up Azure Linux Core 2.0
```
Comment (jslobodzian) [inline SPECS-EXTENDED/cassandra/Readme:5]: Remove the 2.0

File: SPECS-EXTENDED/rubygem-maruku/rubygem-maruku.spec [hunk 2]
```diff
@@ -81,7 +81,7 @@ popd
 - License verified
 
 * Fri Oct 15 2021 Pawel Winogrodzki <pawelwi@microsoft.com> - 0.7.2-10
-- Initial CBL-Mariner import from Fedora 32 (license: MIT).
+- Initial CBL-Azure Linux import from Fedora 32 (license: MIT).
```
Comment (jslobodzian) [inline SPECS-EXTENDED/rubygem-maruku/rubygem-maruku.spec:84]: Revert this one

File: SPECS/calamares/calamares.spec [hunk 3]
```diff
@@ -221,6 +221,9 @@ install -p -m 644 %{SOURCE53} %{buildroot}%{_sysconfdir}/calamares/azl-eula
 %{_libdir}/libcalamaresui.so
 
 %changelog
+* Thu Mar 07 2024 Mykhailo Bykhovtsev <mbykhovtsev@microsoft.com> - 3.3.1-2
```
Comment (jslobodzian) [inline SPECS/calamares/calamares.spec:224]: Note Sam made changes here.

File: SPECS/reaper/Readme [hunk 4]
```diff
@@ -9,7 +9,7 @@
 # d. /usr/local - Current nodejs/npm versions cause build to fail due to incompatible node-sass/node-gyp version dependencies added in reaper's sources - package.json. This requires downgraded nodejs version of 14.18.0 and corresponding npm v6 that needs to be installed.
 # e. <sources>/src/ui/bower_components - bower artifacts built in sources directory
 # f. <sources>/src/ui/node_modules - npm-install related modeuls built in sources directory
-1. Boot up Mariner Core 2.0
+1. Boot up Azure Linux Core 2.0
```
Comment (jslobodzian) [inline SPECS/reaper/Readme:12]: Remove the 2.0

File: toolkit/docs/building/building.md [hunk 5]
```diff
@@ -214,13 +214,13 @@ All images are generated in the `out/images` folder.
...
-# To build a Mariner VHD Image (VHD folder: ../out/images/core-legacy)
+# To build a Azure Linux VHD Image (VHD folder: ../out/images/core-legacy)
 sudo make image CONFIG_FILE=./imageconfigs/core-legacy.json REBUILD_TOOLS=y
 
-# To build a Mariner VHDX Image (VHDX folder ../out/images/core-efi)
+# To build a Azure Linux VHDX Image (VHDX folder ../out/images/core-efi)
 sudo make image CONFIG_FILE=./imageconfigs/core-efi.json REBUILD_TOOLS=y
 
-# To build a core Mariner Contianer (Container Folder: ../out/images/core-container/*.tar.gz
+# To build a core Azure Linux Contianer (Container Folder: ../out/images/core-container/*.tar.gz
```
Comment (jslobodzian) [inline toolkit/docs/building/building.md:223]: replace "a" with "an" for the 3 changes.
Correct spelling of "container"

File: toolkit/docs/building/prerequisites-mariner.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
 
-# Build Requirements on `Mariner`
+# Build Requirements on `Azure Linux`
 
-## Requirements were validated on `Mariner 2.0`
+## Requirements were validated on `Azure Linux 2.0`
```
Comment (jslobodzian) [inline toolkit/docs/building/prerequisites-mariner.md:4]: Replace 2.0 with 3.0?  or Drop 2.0.

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg [hunk 7]
```diff
@@ -72,12 +72,12 @@ cloud_final_modules:
 # (not accessible to handlers/transforms)
 system_info:
    # This will affect which distro class gets used
-   distro: mariner
+   distro: azl
```
Comment (jslobodzian) [inline toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:75]: azurelinux

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg [hunk 8]
```diff
@@ -72,12 +72,12 @@ cloud_final_modules:
...
    # Default user name + that default users groups (if added/used)
    default_user:
-     name: mariner
+     name: azl
```
Comment (jslobodzian) [inline toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:78]: azl_user

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg [hunk 9]
```diff
@@ -72,12 +72,12 @@ cloud_final_modules:
...
      lock_passwd: true
-     gecos: Mariner
+     gecos: azl
```
Comment (jslobodzian) [inline toolkit/imageconfigs/additionalconfigs/cloud-init.cfg:80]: AzureLinux

File: toolkit/resources/assets/isomaker/iso_root_static_files/boot/grub2/grub.cfg [hunk 10]
```diff
@@ -2,7 +2,7 @@ set timeout=0
 
 # The use of mariner.media=CDROM is a workaround that our installer does not require
 # but it is observed to be needed to boot on some hardware/SoCs.
-menuentry "CBL-Mariner" {
+menuentry "Azure Linux" {
```
Comment (jslobodzian) [inline toolkit/resources/assets/isomaker/iso_root_static_files/boot/grub2/grub.cfg:5]: Ask @Chris Gunn or @Gary Swalling what is best here.

File: toolkit/tools/grapher/grapher.go [hunk 11]
```diff
@@ -91,7 +91,7 @@ func main() {
 			enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagPreview
 		}
 		if *disableDefaultRepos {
-			enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagMarinerDefaults
+			enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagAzureLinuxDefaults
```
Comment (jslobodzian) [inline toolkit/tools/grapher/grapher.go:94]: Consider "....DistroDefaults" and drop the AzureLinux

File: toolkit/tools/imagecustomizer/README.md [hunk 12]
```diff
@@ -1,11 +1,11 @@
-# Mariner Image Customizer
+# Azure Linux Image Customizer
 
-The Mariner Image Customizer (MIC) is a tool that can take an existing generic Mariner
+The Azure Linux Image Customizer (MIC) is a tool that can take an existing generic Azure Linux
```
Comment (jslobodzian) [inline toolkit/tools/imagecustomizer/README.md:3]: Check with Chris Gunn on what he wants to do here.

File: toolkit/tools/imagecustomizer/docs/iso.md [hunk 13]
```diff
@@ -1,8 +1,8 @@
-# Mariner Image Customizer ISO Support
+# Azure Linux Image Customizer ISO Support
 
 ## Overview
 
-Given a full disk image, the Mariner Image Customizer (MIC) can generate a
+Given a full disk image, the Azure Linux Image Customizer (MIC) can generate a
```
Comment (jslobodzian) [inline toolkit/tools/imagecustomizer/docs/iso.md:5]: And here.  MIC-> ALIC?

File: toolkit/tools/imagegen/attendedinstaller/speakuputils/speakuputils.go [hunk 14]
```diff
@@ -18,7 +18,7 @@ const (
 
 // CreateVirtualKeyboard creates and returns a virtual keyboard from the uinput package
 func CreateVirtualKeyboard() (keyboard uinput.Keyboard, err error) {
-	keyboard, err = uinput.CreateKeyboard("/dev/uinput", []byte("MarinerVirtualKeyboard"))
+	keyboard, err = uinput.CreateKeyboard("/dev/uinput", []byte("AzureLinuxVirtualKeyboard"))
```
Comment (jslobodzian) [inline toolkit/tools/imagegen/attendedinstaller/speakuputils/speakuputils.go:21]: consider changing to "myLinuxVirtualKeyboard" or "isoVirtualKeyboard"

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 15]
```diff
@@ -1147,7 +1147,7 @@ func InstallGrubDefaults(installRoot, rootDevice, bootUUID, bootPrefix string,
 	}
 
 	if includeLegacyCfg {
-		// Add the legacy /boot/grub2/grub.cfg file, which was used in Mariner 2.0.
+		// Add the legacy /boot/grub2/grub.cfg file, which was used in Azure Linux 2.0.
```
Comment (jslobodzian) [inline toolkit/tools/imagegen/installutils/installutils.go:1150]: drop the 2.0

File: toolkit/tools/imagepkgfetcher/imagepkgfetcher.go [hunk 16]
```diff
@@ -88,7 +88,7 @@ func main() {
 		enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagUpstream
 	}
 	if *disableDefaultRepos {
-		enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagMarinerDefaults
+		enabledRepos = enabledRepos & ^rpmrepocloner.RepoFlagAzureLinuxDefaults
```
Comment (jslobodzian) [inline toolkit/tools/imagepkgfetcher/imagepkgfetcher.go:91]: consider replacing AzureLinux with Distro here (if you do it in the other places)

File: toolkit/tools/liveinstaller/liveinstaller.go [hunk 17]
```diff
@@ -167,22 +167,22 @@ func runBootEntryCreationCommand(installDetails installationDetails) (err error)
...
-func removeOldMarinerBootTargets() (err error) {
+func removeOldAzureLinuxBootTargets() (err error) {
 	const squashErrors = false
-	logger.Log.Info("Removing pre-existing 'Mariner' boot targets from efibootmgr")
+	logger.Log.Info("Removing pre-existing 'Azure Linux' boot targets from efibootmgr")
 	program := "efibootmgr" // Default behavior when piped or called without options is to print current boot order in a human-readable format
 	commandArgs := []string{
-		"|", "grep", "\"Mariner\"", // Filter boot order for Mariner boot targets
-		"|", "sed", "'s/* Mariner//g'", // Pruning for just the bootnum
+		"|", "grep", "\"Azure Linux\"", // Filter boot order for Azure Linux boot targets
```
Comment (jslobodzian) [inline toolkit/tools/liveinstaller/liveinstaller.go:182]: check with Chris Co

File: toolkit/tools/pkggen/worker/create_worker_chroot.sh [hunk 18]
```diff
@@ -79,7 +79,7 @@ while read -r package || [ -n "$package" ]; do
     install_one_toolchain_rpm "$package"
 done < "$packages"
 
-# If the host machine rpm version is >= 4.16 (such as Mariner 2.0), it will create an "sqlite" rpm database backend incompatible with Mariner 1.0 (which uses "bdb")
+# If the host machine rpm version is >= 4.16 (such as Azure Linux 2.0), it will create an "sqlite" rpm database backend incompatible with Azure Linux 1.0 (which uses "bdb")
```
Comment (jslobodzian) [inline toolkit/tools/pkggen/worker/create_worker_chroot.sh:82]: restore this, and then add Azure Linux 3.0

such as Mariner 2.0 or Azure Linux 3.0

File: SPECS/grub2/sbat.csv.in [hunk 19]
```diff
@@ -1,3 +1,3 @@
 sbat,1,SBAT Version,sbat,1,https://github.com/rhboot/shim/blob/main/SBAT.md
 grub,2,Free Software Foundation,grub,@@VERSION@@,https://www.gnu.org/software/grub/
-grub.mariner,2,Microsoft,grub2,@@VERSION_RELEASE@@,https://github.com/microsoft/CBL-Mariner
+grub.mariner,2,Microsoft,grub2,@@VERSION_RELEASE@@,https://github.com/microsoft/azurelinux
```
Comment (christopherco) [inline SPECS/grub2/sbat.csv.in:3]:
```suggestion
grub.azurelinux,2,Microsoft,grub2,@@VERSION_RELEASE@@,https://github.com/microsoft/azurelinux
```

File: SPECS/grub2/grub2.spec [hunk 20]
```diff
@@ -387,6 +387,9 @@ cp $GRUB_PXE_MODULE_SOURCE $EFI_BOOT_DIR/$GRUB_PXE_MODULE_NAME
 %config(noreplace) %{_sysconfdir}/grub.d/41_custom
 
 %changelog
+* Wed Feb 28 2024 Mykhailo Bykhovtsev <mbykhovtsev@microsoft.com> - 2.06-15
+- Updated sbat.csv.in to replace github link with new.
```
Comment (christopherco) [inline SPECS/grub2/grub2.spec:391]:
```suggestion
- Updated sbat.csv.in to reflect new distro name.
```

File: SPECS/shim-unsigned-aarch64/sbat.csv.in [hunk 21]
```diff
@@ -1 +1 @@
-shim.mariner,1,Microsoft,shim,@@VERSION_RELEASE@@,https://github.com/microsoft/CBL-Mariner
+shim.mariner,1,Microsoft,shim,@@VERSION_RELEASE@@,https://github.com/microsoft/azurelinux
```
Comment (christopherco) [inline SPECS/shim-unsigned-aarch64/sbat.csv.in:1]:
```suggestion
shim.azurelinux,1,Microsoft,shim,@@VERSION_RELEASE@@,https://github.com/microsoft/azurelinux
```

File: SPECS/shim-unsigned-x64/sbat.csv.in [hunk 22]
```diff
@@ -1 +1 @@
-shim.mariner,1,Microsoft,shim,@@VERSION_RELEASE@@,https://github.com/microsoft/CBL-Mariner
+shim.mariner,1,Microsoft,shim,@@VERSION_RELEASE@@,https://github.com/microsoft/azurelinux
```
Comment (christopherco) [inline SPECS/shim-unsigned-x64/sbat.csv.in:1]:
```suggestion
shim.azurelinux,1,Microsoft,shim,@@VERSION_RELEASE@@,https://github.com/microsoft/azurelinux
```

**Review by christopherco — CHANGES_REQUESTED** [review]

File: toolkit/resources/assets/isomaker/iso_root_static_files/boot/grub2/grub.cfg (hunk 10 above)
Comment (christopherco) [reply toolkit/resources/assets/isomaker/iso_root_static_files/boot/grub2/grub.cfg:5]: This change looks good - it changes grub's boot menu entry string that is displayed for this entry

File: README.md [hunk 23]
```diff
@@ -21,8 +21,8 @@ Instructions for building Azure Linux may be found here: [Toolkit Documentation]
 
 You can try Azure Linux with the following ISO images:
 
-- [Mariner 2.0 x86_64 ISO](https://aka.ms/mariner-2.0-x86_64-iso).
-- [Mariner 1.0 x86_64 ISO](https://aka.ms/mariner-1.0-x86_64-iso).
+- [Azure Linux 2.0 x86_64 ISO](https://aka.ms/mariner-2.0-x86_64-iso).
+- [Azure Linux 1.0 x86_64 ISO](https://aka.ms/mariner-1.0-x86_64-iso).
```
[the remark this answers is not in the document]
Comment (mbykhovtsev-ms) [reply README.md]: Updated

File: SPECS-EXTENDED/cassandra/Readme (hunk 1 above)
Comment (mbykhovtsev-ms) [reply SPECS-EXTENDED/cassandra/Readme]: Updated

File: SPECS/calamares/calamares.spec (hunk 3 above)
Comment (mbykhovtsev-ms) [reply SPECS/calamares/calamares.spec:224]: Updated

File: SPECS/ostree/0003-ostree-converting-osname-to-mariner.patch [hunk 24]
```diff
@@ -26,7 +26,7 @@ index 976c4b5..263e2f5 100644
      }
  
 -  /* Copying Photon Specific config files to sutiable location */
-+  /* Copying Mariner Specific config files to sutiable location */
++  /* Copying Azure Linux Specific config files to sutiable location */
```
[the remark this answers is not in the document]
Comment (mbykhovtsev-ms) [reply SPECS/ostree/0003-ostree-converting-osname-to-mariner.patch]: Updated

File: SPECS/reaper/Readme (hunk 4 above)
Comment (mbykhovtsev-ms) [reply SPECS/reaper/Readme]: Updated

File: toolkit/docs/building/prerequisites-mariner.md (hunk 6 above)
Comment (mbykhovtsev-ms) [reply toolkit/docs/building/prerequisites-mariner.md]: Updated

File: toolkit/docs/building/building.md (hunk 5 above)
Comment (mbykhovtsev-ms) [reply toolkit/docs/building/building.md]: Updated all occurences

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg (hunk 7 above)
Comment (mbykhovtsev-ms) [reply toolkit/imageconfigs/additionalconfigs/cloud-init.cfg]: Updated

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg (hunk 9 above)
Comment (mbykhovtsev-ms) [reply toolkit/imageconfigs/additionalconfigs/cloud-init.cfg]: Updated

File: toolkit/resources/assets/isomaker/iso_root_static_files/boot/grub2/grub.cfg (hunk 10 above)
Comment (mbykhovtsev-ms) [reply toolkit/resources/assets/isomaker/iso_root_static_files/boot/grub2/grub.cfg:5]: Thanks for verifying

File: toolkit/tools/grapher/grapher.go (hunk 11 above)
Comment (mbykhovtsev-ms) [reply toolkit/tools/grapher/grapher.go]: Updated

File: toolkit/tools/imagecustomizer/README.md (hunk 12 above)
Comment (mbykhovtsev-ms) [reply toolkit/tools/imagecustomizer/README.md]: Checked with Chris, ALIC is not good acronym. Decided to drop use of acronym everywhere

File: toolkit/tools/imagecustomizer/docs/iso.md (hunk 13 above)
Comment (mbykhovtsev-ms) [reply toolkit/tools/imagecustomizer/docs/iso.md]: Checked with Chris, ALIC is not good acronym. Decided to drop use of acronym everywhere

File: toolkit/tools/imagegen/attendedinstaller/speakuputils/speakuputils.go (hunk 14 above)
Comment (mbykhovtsev-ms) [reply toolkit/tools/imagegen/attendedinstaller/speakuputils/speakuputils.go]: Updated

File: toolkit/tools/imagegen/installutils/installutils.go (hunk 15 above)
Comment (mbykhovtsev-ms) [reply toolkit/tools/imagegen/installutils/installutils.go:1150]: Updated

File: toolkit/tools/pkggen/worker/create_worker_chroot.sh (hunk 18 above)
Comment (mbykhovtsev-ms) [reply toolkit/tools/pkggen/worker/create_worker_chroot.sh]: Updated

File: SPECS/grub2/sbat.csv.in (hunk 19 above)
Comment (mbykhovtsev-ms) [reply SPECS/grub2/sbat.csv.in]: Updated

File: SPECS/shim-unsigned-aarch64/sbat.csv.in (hunk 21 above)
Comment (mbykhovtsev-ms) [reply SPECS/shim-unsigned-aarch64/sbat.csv.in]: Updated

File: SPECS/shim-unsigned-x64/sbat.csv.in (hunk 22 above)
Comment (mbykhovtsev-ms) [reply SPECS/shim-unsigned-x64/sbat.csv.in]: Updated

File: SPECS-EXTENDED/rubygem-maruku/rubygem-maruku.spec (hunk 2 above)
Comment (mbykhovtsev-ms) [reply SPECS-EXTENDED/rubygem-maruku/rubygem-maruku.spec]: Updated

File: toolkit/imageconfigs/additionalconfigs/cloud-init.cfg (hunk 8 above)
Comment (mbykhovtsev-ms) [reply toolkit/imageconfigs/additionalconfigs/cloud-init.cfg]: Updated

File: SPECS/grub2/grub2.spec (hunk 20 above)
Comment (mbykhovtsev-ms) [reply SPECS/grub2/grub2.spec]: Updated

**Review by jslobodzian — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**